### ISSUE TITLE
In 4.6.0 multiple ports connection cannot find primary node

### ISSUE THREAD

**Issue by Marchys** [Automattic/mongoose#4517]
see the example:
`mongodb://host1:port1,host2:port2,host3:port3/database` in 4.5.10 works
`mongodb://host1:port1,host2:port2,host3:port3/database` in 4.6.0 doesn't work
error: 
`MongoDB connection error: { [MongoError: no primary found in replicaset]
  name: 'MongoError',
  message: 'no primary found in replicaset' }`

**Comment by dmkelly** [comment]
I also ran into this issue on 4.6.0. Switching back to 4.5.5 worked for me.

My connection string looks like the following:

```
mongodb://host0:27017,host1:27017,host2:27017/database?replicaSet=name
```

**Comment by twalling** [comment]
Looks like this may be addressed with mongo driver 2.2.10

https://github.com/mongodb/node-mongodb-native/blob/2.2/HISTORY.md#2210-2016-09-15

references bug
https://jira.mongodb.org/browse/NODE-818

**Comment by vkarpov15** [comment]
Will upgrade to 2.2.10 for next release
